src/testhost.x86/TestHostTraceListener.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+namespace Microsoft.VisualStudio.TestPlatform.TestHost
+{
+#if !NET451
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using System;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Reflection;
+
+    internal class TestHostTraceListener : TraceListener
+    {
+        public static void Setup()
+        {
+            EqtTrace.Info("Setting up debug trace listener.");
+            EqtTrace.Verbose("TestPlatformTraceListener.Setup: Removing listener {0}.", Trace.Listeners[0]);
+            Trace.Listeners[0] = new TestHostTraceListener();
+            EqtTrace.Verbose("TestPlatformTraceListener.Setup: Added test platform trace listener.");
+
+#if NETCOREAPP2_1
+            try
+            {
+                // workaround for netcoreapp2.1 where the trace listener api is not called when 
+                // Debug.Assert fails. This method is internal, but the class is on purpose keeping the 
+                // callback settable so tests can set the callback
+                var field = typeof(Debug).GetField("s_ShowDialog", BindingFlags.Static | BindingFlags.NonPublic);
+                var value = field.GetValue(null);
+                field.SetValue(null, (Action<string, string, string, string>)ShowDialog);
```

Comment:
This is interesting 😄

Reply:
Nevermind - I missed the fact that the workaround was specifically for netcoreapp2.1 and no other version.